src/gp_activetable.c

```diff
@@ -1133,16 +1082,20 @@ pull_active_list_from_seg(void)
  * table size on all of the segments.
  */
 static void
-pull_active_table_size_from_seg(HTAB *local_table_stats_map, char *active_oid_array)
+pull_active_table_size_from_seg(const char *active_oids)
 {
-	CdbPgResults   cdb_pgresults = {NULL, 0};
-	StringInfoData sql_command;
-	int            i;
-	int            j;
+	ActiveTableEntry *entry;
+	CdbPgResults      cdb_pgresults = {NULL, 0};
+	StringInfoData    sql_command;
+	int               i;
+	int               j;
+	HASHCTL ctl = {.keysize = sizeof(Oid), .entrysize = sizeof(ActiveTableEntry), .hcxt = CurrentMemoryContext};
+	HTAB   *local_table_stats_map = diskquota_hash_create("local active table map with size info", 1024, &ctl,
```

Comment:
And due to what we decrease memory consumption? For is_init case we succeeded to get rid of  hash map, but here we inevitably use it. Map ActiveTableEntry is simplier and the map lives shorter

Reply:
> And due to what we decrease memory consumption? For is_init case we succeeded to get rid of hash map, but here we inevitably use it.

We significantly reduce the memory consumption during initialization (which was the original problem), and also slightly reduce the memory consumption during normal operation, due to the fact that the local hashmap stores the size only from the coordinator instead of the sizes from all segments.

Reply:
> What about leaving the general flow uncahnged for normal operation case?

This is not possible because now `calculate_table_disk_usage` knows nothing about active tables.

Reply:
>This is not possible because now calculate_table_disk_usage

Let use the hash map for normal operation mode as before changes, for is_init mode just do not update the quotas

Reply:
> use the hash map for normal operation mode as before changes

Why do we need a long-lived hashmap with active tables when we only need a long-lived serialized array with active tables?

Reply:
decentralized update really hits the code readability